src/mapgen.cpp

```diff
@@ -663,7 +663,35 @@ class jmapgen_sign : public jmapgen_piece {
         const int ry = y.get();
         m.furn_set( rx, ry, f_null );
         m.furn_set( rx, ry, "f_sign" );
-        m.set_signage( tripoint( rx, ry, m.get_abs_sub().z ), signage );
+
+        tripoint abs_sub = m.get_abs_sub();
+        tripoint abs_omt = overmap_buffer.sm_to_omt_copy(abs_sub);
+
+        std::string signtext;
+
+        if( !snippet.empty() ) {
+            // select a snippet from the category
+            signtext = SNIPPET.get( SNIPPET.assign( snippet ) );
+        } else if( !signage.empty() ) {
+            signtext = signage;
+        } else {
+            // throw some sort of error
+            debugmsg("jmapgen_sign with no signage or snippet defined");
```

Comment:
That should probably be in the constructor. The failure is easy to detect, and it should be shown as soon as possible. You can even use the json error reporting from there: `jsi.throw_error("need either X or Y, but found neither")`